api/clients/v2/disperser_client.go

```diff
@@ -128,42 +145,67 @@ func (c *disperserClient) DisperseBlob(
 	blobVersion corev2.BlobVersion,
 	quorums []core.QuorumID,
 ) (*dispv2.BlobStatus, corev2.BlobKey, error) {
-	c.requestMutex.Lock()
-	defer c.requestMutex.Unlock()
+	return c.DisperseBlobWithProbe(ctx, data, blobVersion, quorums, nil)
+}
 
-	err := c.initOnceGrpcConnection()
+// DisperseBlobWithProbe disperses a blob with the given data, blob version, and quorums. If sequenceProbe is not nil,
+// the probe is used to capture metrics during the dispersal process.
+func (c *disperserClient) DisperseBlobWithProbe(
+	ctx context.Context,
+	data []byte,
+	blobVersion corev2.BlobVersion,
+	quorums []core.QuorumID,
+	probe *common.SequenceProbe,
+) (*dispv2.BlobStatus, corev2.BlobKey, error) {
+
+	if len(quorums) == 0 {
+		return nil, [32]byte{}, api.NewErrorInvalidArg("quorum numbers must be provided")
+	}
+	if c.signer == nil {
+		return nil, [32]byte{}, api.NewErrorInternal("uninitialized signer for authenticated dispersal")
+	}
+	for _, q := range quorums {
+		if q > corev2.MaxQuorumID {
+			return nil, [32]byte{}, api.NewErrorInvalidArg("quorum number must be less than 256")
+		}
+	}
+
+	probe.SetStage("acquire_accountant_lock")
+	c.accountantLock.Lock()
+
+	probe.SetStage("prepare_for_dispersal")
+
+	err := c.initOncePopulateAccountant(ctx)
 	if err != nil {
 		return nil, [32]byte{}, api.NewErrorFailover(err)
 	}
-	err = c.initOncePopulateAccountant(ctx)
+	err = c.initOnceGrpcConnection()
 	if err != nil {
 		return nil, [32]byte{}, api.NewErrorFailover(err)
 	}
 
-	if c.signer == nil {
-		return nil, [32]byte{}, api.NewErrorInternal("uninitialized signer for authenticated dispersal")
-	}
-
 	symbolLength := encoding.GetBlobLengthPowerOf2(uint(len(data)))
 	payment, err := c.accountant.AccountBlob(ctx, time.Now().UnixNano(), uint64(symbolLength), quorums)
 	if err != nil {
+		c.accountantLock.Unlock()
 		return nil, [32]byte{}, fmt.Errorf("error accounting blob: %w", err)
 	}
 
-	if len(quorums) == 0 {
-		return nil, [32]byte{}, api.NewErrorInvalidArg("quorum numbers must be provided")
-	}
-
-	for _, q := range quorums {
-		if q > corev2.MaxQuorumID {
-			return nil, [32]byte{}, api.NewErrorInvalidArg("quorum number must be less than 256")
-		}
+	if payment.CumulativePayment == nil || payment.CumulativePayment.Sign() == 0 {
+		// This request is using reserved bandwidth, no need to prevent parallel dispersal.
+		c.accountantLock.Unlock()
+	} else {
+		// This request is using on-demand bandwidth, current implementation requires sequential dispersal.
+		defer c.accountantLock.Unlock()
```

Comment:
Just a thought- if this ends up presenting a performance issue, I think some easy optimization could be done to minimize lock time: some of the payload/blob/commitment work is quite expensive, and could be done prior to acquiring the accountant lock

Reply:
Good point. In the initial metrics I've collected, the time spent in this part of the code is very small. But if it ever starts to become an issue, we can circle back on it.

Reply:
Do any of your tests use on-demand payments?

Reply:
no, I'm unsure how to configure on-demand payments

Reply:
Resolving for now